### PR TITLE
Add NSFW flag to Slash Commands

### PR DESCRIPTION
**Pull request by DubyaDude**
# Summary
Discord recently announced the scanning of attachments on Slash Commands. Setting the slash command as NSFW would allow attachments to not be scanned. This PR adds the NSFW flag to Create/Edit slash command requests.

DiscordCanary_XHrKcZAE5J

# Notes
I went about adding the NSFW tag through the `SlashCommandAttribute` constructor. I wasn't sure if it'd be preferred to do it via its own attribute. If so, I can change that.

### CODE REVIEW

**Comment by DubyaDude** [comment]
Need to test, will do it tomorrow.

**Review by InFTord — APPROVED** [review]
LGTM

**Comment by Plerx2493** [comment]
I am not totaly sure but as far as i understand the slash command libary, you forgot to add the required logic in `SlashCommandsExtension` in the registerCommands section

**Comment by DubyaDude** [comment]
> I am not totaly sure but as far as i understand the slash command libary, you forgot to add the required logic in `SlashCommandsExtension` in the registerCommands section

Should be set now, let me know if I missed a spot.

**Comment by Plerx2493** [comment]
LGTM

**Comment by DubyaDude** [comment]
Works, good to merge.

File: DSharpPlus/Entities/Interaction/Application/DiscordApplicationCommand.cs [hunk 1]
```diff
@@ -115,7 +120,7 @@ public string Mention
         /// <param name="description_localizations">Localization dictionary for <paramref name="description"/> field. Values follow the same restrictions as <paramref name="description"/>.</param>
         /// <param name="allowDMUsage">Whether this command can be invoked in DMs.</param>
         /// <param name="defaultMemberPermissions">What permissions this command requires to be invoked.</param>
```
Comment (Plerx2493) [inline DSharpPlus/Entities/Interaction/Application/DiscordApplicationCommand.cs:122]: documentation

**Pull request merged**